swagger.json

```diff
@@ -3105,6 +3168,76 @@
                 }
             }
         },
+        "VehicleStats": {
+            "type": "object",
+            "description": "Engine state and aux input data for a vehicle.",
+            "properties": {
+                "engineState": {
+                    "$ref": "#/definitions/ObjectStatEngineState"                    
+                },
+                "auxInput1": {
+                    "type": "object",
+                    "required": ["name", "values"],
+                    "properties": {
+                        "name": {
+                            "type": "string",
+                            "example": "Boom"
+                        },
+                        "values": {
+                            "$ref": "#/definitions/ObjectStatIntSeries"
+                        }
+                    }
+                },
+                "auxInput2": {
+                    "type": "object",
+                    "required": ["name", "values"],
+                    "properties": {
+                        "name": {
+                            "type": "string",
+                            "example": "Stop Paddle"
+                        },
+                        "values": {
+                            "$ref": "#/definitions/ObjectStatIntSeries"
+                        }
+                    }
+                }
+            }
+        },
+        "ObjectStatInt": {
+            "type": "object",
+            "description": "A single datapoint of timeseries data with an integer value.",
+            "required": ["timeMs", "value"],
+            "properties": {
+                "timeMs": {
+                    "type": "integer",
+                    "description": "Timestamp in Unix epoch milliseconds.",
+                    "format": "int64",
+                    "example": "1546542978484"
+                },
+                "value": {
+                    "type": "integer",
+                    "description": "Integer value.",
+                    "format": "int64",
+                    "example": "1030"
+                }
+            }       
+        },
+        "ObjectStatIntSeries": {
+            "type": "array",
+            "items": {
+                "$ref": "#/definitions/ObjectStatInt"
+            }
+        },
+        "ObjectStatEngineState": {
+            "type": "object",
+            "description": "The state of the vehicle over time. 0 = Off, 1 = Running, 2 = Idle.",
```

Comment:
Could this be a swagger enum? https://swagger.io/docs/specification/2-0/enums/

Could we also use the human readable name rather than the int value?

Reply:
Yep, changed to be an enum of human-readable values.